### ISSUE TITLE
Not working on M1 Mac

### ISSUE THREAD

**Issue by louwers** [Mateusz-Grzelinski/actionlint-py#9]
Seems like a reasonable platform to support.

Interested in this because I can't find a way to use pre-commit with actionlint and shellcheck.

```
[INFO] Initializing environment for https://github.com/Mateusz-Grzelinski/actionlint-py.
[INFO] Installing environment for https://github.com/Mateusz-Grzelinski/actionlint-py.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/bin/python', '-mpip', 'install', '.')
return code: 1
stdout:
    Processing /Users/bart/.cache/pre-commit/repo0xe4ef3u
      Preparing metadata (setup.py): started
      Preparing metadata (setup.py): finished with status 'done'
    Building wheels for collected packages: actionlint-py
      Building wheel for actionlint-py (setup.py): started
      Building wheel for actionlint-py (setup.py): finished with status 'error'
      Running setup.py clean for actionlint-py
    Failed to build actionlint-py
stderr:
      error: subprocess-exited-with-error
      
      × python setup.py bdist_wheel did not run successfully.
      │ exit code: 1
      ╰─> [63 lines of output]
          /Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/setuptools/config/setupcfg.py:293: _DeprecatedConfig: Deprecated config in `setup.cfg`
          !!
          
                  ********************************************************************************
                  The license_file parameter is deprecated, use license_files instead.
          
                  By 2023-Oct-30, you need to update your project and remove deprecated calls
                  or your builds will no longer be supported.
          
                  See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
                  ********************************************************************************
          
          !!
            parsed = self.parsers.get(option_name, lambda x: x)(value)
          running bdist_wheel
          running build
          running build_py
          creating build
          creating build/lib
          copying setup_auto_update.py -> build/lib
          running fetch_binaries
          Traceback (most recent call last):
            File "<string>", line 2, in <module>
            File "<pip-setuptools-caller>", line 34, in <module>
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/setup.py", line 166, in <module>
              setup(version=f'{ACTIONLINT_VERSION}.{PY_VERSION}', cmdclass=command_overrides)
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/setuptools/__init__.py", line 107, in setup
              return distutils.core.setup(**attrs)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
              return run_commands(dist)
                     ^^^^^^^^^^^^^^^^^^
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
              dist.run_commands()
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
              self.run_command(cmd)
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/setuptools/dist.py", line 1234, in run_command
              super().run_command(command)
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
              cmd_obj.run()
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/wheel/bdist_wheel.py", line 343, in run
              self.run_command("build")
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
              self.distribution.run_command(command)
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/setuptools/dist.py", line 1234, in run_command
              super().run_command(command)
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
              cmd_obj.run()
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/setuptools/_distutils/command/build.py", line 131, in run
              self.run_command(cmd_name)
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
              self.distribution.run_command(command)
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/setuptools/dist.py", line 1234, in run_command
              super().run_command(command)
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/py_env-python3.11/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
              cmd_obj.run()
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/setup.py", line 112, in run
              url, sha256 = get_download_url()
                            ^^^^^^^^^^^^^^^^^^
            File "/Users/bart/.cache/pre-commit/repo0xe4ef3u/setup.py", line 39, in get_download_url
              postfix = POSTFIX_SHA256[(sys.platform, platform.machine())]
                        ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
          KeyError: ('darwin', 'arm64')
          [end of output]
      
      note: This error originates from a subprocess, and is likely not a problem with pip.
      ERROR: Failed building wheel for actionlint-py
    ERROR: Could not build wheels for actionlint-py, which is required to install pyproject.toml-based projects
Check the log at /Users/bart/.cache/pre-commit/pre-commit.log
```

**Comment by Mateusz-Grzelinski** [comment]
oh, the mac is arm architecture, of course. Totaly forgot about it.

**Comment by Mateusz-Grzelinski** [comment]
please check if this ends with success - I do not have any way to test mac arm. Gihub exposes only x64 mac runners After you confirm I will trigger release (#10)
```
pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ actionlint-py==1.6.25.4.dev.2
```

**Comment by louwers** [comment]
@Mateusz-Grzelinski 
```
% pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ actionlint-py==1.6.25.4.dev.2
zsh: command not found: pip
bart@Barts-MacBook-Air maplibre-native % pip3 install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ actionlint-py==1.6.25.4.dev.2
Looking in indexes: https://test.pypi.org/simple/, https://pypi.org/simple/
Collecting actionlint-py==1.6.25.4.dev.2
  Downloading https://test-files.pythonhosted.org/packages/6a/28/62da31dc0acc92808eff6a8c2c2b62a354e71adba9f8a138b10ea6c1925f/actionlint_py-1.6.25.4.dev2.tar.gz (9.4 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: actionlint-py
  Building wheel for actionlint-py (pyproject.toml) ... done
  Created wheel for actionlint-py: filename=actionlint_py-1.6.25.4.dev2-py2.py3-none-macosx_13_0_arm64.whl size=1749406 sha256=d6f7c53683f0a0bab939f277f007b20acfff0408288c4790b6f1b6850b45af89
  Stored in directory: /Users/bart/Library/Caches/pip/wheels/4f/e0/32/952fd3ec325d10403ac835dfc38d94f88874af7637bcab1a2c
Successfully built actionlint-py
Installing collected packages: actionlint-py
Successfully installed actionlint-py-1.6.25.4.dev2
```

**Comment by Mateusz-Grzelinski** [comment]
Looks good. I just hope the executable is correct

**Comment by Mateusz-Grzelinski** [comment]
new version will work for you
```
pip install actionlint-py==1.6.25.8
```
or just
```
 pre-commit autoupdate
```

**Comment by louwers** [comment]
Getting a different error this time around:
```
[Errno 8] Exec format error: '/Users/bart/.cache/pre-commit/repotxsqggc5/py_env-python3.11/bin/actionlint'
```

Installed it like this:

```
- repo: https://github.com/Mateusz-Grzelinski/actionlint-py
  rev: v1.6.25.8
```

**Comment by Mateusz-Grzelinski** [comment]
that was very stupid mistake of mine. Try version `v1.6.25.9`
note new section in readme about shellcheck-py as dependency

**Comment by Mateusz-Grzelinski** [comment]
I just read your first comment: xd

> I can't find a way to use pre-commit with actionlint and shellcheck

Normally you would need to install actionlint and shellcheck in your system and then use `repo local` and `language: system` to run executable directly.
```
- repo: local
  hooks:
    - id: action
      name: Local Actionlint
      entry: actionlint
      language: system
      types: [ "yaml" ]
      files: "^.github/workflows/"
```

**Comment by louwers** [comment]
@Mateusz-Grzelinski I don't want to use local because then different contributors will have different versions installed, leading to hard to debug conflicts. Also, that does not work with pre-commit.ci, which we use.

Let me try v1.6.25.9

**Comment by louwers** [comment]
Works.

Thanks!
